app/code/Magento/Bundle/Model/ResourceModel/Option.php

```diff
@@ -81,30 +81,44 @@ protected function _afterSave(\Magento\Framework\Model\AbstractModel $object)
     {
         parent::_afterSave($object);
 
-        $condition = [
+        $conditions = [
             'option_id = ?' => $object->getId(),
-            'store_id = ? OR store_id = 0' => $object->getStoreId(),
+            'store_id = ?' => $object->getStoreId(),
```

Comment:
Is there any chance that the `getStoreId` is not set? In this case if it is not set what would happen here?